src/transformers/generation_tf_utils.py

```diff
@@ -1973,36 +1974,36 @@ def greedy_search(
         decoder_hidden_states = [] if (return_dict_in_generate and output_hidden_states) else None
 
         # 3. init tensors to use for "xla-compileable" generate function
-        # define bsz, seq_length
-        batch_size, seq_length = input_ids.shape
+        batch_size, cur_len = input_ids.shape
 
-        # initialize `generated`, `finished_sequences`, and `current_pos`
+        # initialize `generated` (pre-populated with `pad_token_id`), `finished_sequences`
         generated = tf.TensorArray(
             element_shape=(batch_size,),
             dtype=tf.int32,
             dynamic_size=False,
             size=max_length,
             clear_after_read=False,
         )
+        if pad_token_id:  # ignores the cases when it is 0 or None
+            for i in range(max_length):
+                generated = generated.write(i, tf.broadcast_to(pad_token_id, (batch_size,)))
```

Comment:
Since `TensorArray` is write-once, will this not make it impossible to write anything more into `generated`?

Reply:
Interesting! It does say that in the docs, right at the top, but I've seen no limitations with this -- beam search already had this change 🤔 

I can also move this loop to populate any remaining positions with the pad token, it's probably better! WDYT?

Reply:
Ah! Let me run some experiments, then. Maybe if you do this in a graph that never tries to backprop through the array, it lets you do it?

Reply:
Tested and the docs are just entirely wrong, even for `tf.float32` arrays, at least when you're not in a `GradientTape` context!